src/vorta/views/partials/treemodel.py

```diff
@@ -994,3 +1042,246 @@ def lessThan(self, left: QModelIndex, right: QModelIndex) -> bool:
+        parser = super().get_parser()
+        parser.add_argument(
+            "-b", "--balance", type=FileTreeSortFilterProxyModel.valid_size, help="Match by balance size."
+        )
+
+        return parser
+
+    def filterAcceptsRow(self, sourceRow: int, sourceParent: QModelIndex) -> bool:
+        """
+        Return whether the row should be accepted.
+        """
+
+        if not self.searchPattern:
+            return True
+
+        if not super().filterAcceptsRow(sourceRow, sourceParent):
+            return False
+
+        model = self.sourceModel()
+        item = model.index(sourceRow, 0, sourceParent).internalPointer()
+
+        if self.searchPattern.balance:
+            item_balance = item.data.size
+
+            for filter_balance in self.searchPattern.balance:
+                if not compare_values_with_sign(item_balance, filter_balance[1], filter_balance[0]):
+                    return False
+
+        return True
+
+
+class ExtractTreeSortFilterProxyModel(FileTreeSortFilterProxyModel):
+    """
+    Tree model for diff view.
+    """
+
+    def __init__(self, parent=None) -> None:
+        super().__init__(parent)
+
+    @staticmethod
+    def parse_date(value):
+        """
+        Parse the date string into a tuple of two values.
+        """
+
+        comparison_sign = ['<', '>', '<=', '>=', '=']
+
+        if not any(value.startswith(unit) for unit in comparison_sign):
+            raise argparse.ArgumentTypeError("Invalid date format. Supported comparison signs: <, >, <=, >=, =")
+
+        if value[1] == '=':
+            date = value[2:]
+            sign = value[:2]
+        else:
+            date = value[1:]
+            sign = value[:1]
+
+        try:
+            date = datetime.strptime(date, '%Y-%m-%d')
+        except ValueError:
+            raise argparse.ArgumentTypeError("Invalid date format. Must be YYYY-MM-DD.")
+
+        return (sign, date)
+
+    @staticmethod
+    def valid_date_range(value):
+        """Parse the date range string."""
+        date = value.split(',')
+
+        if len(date) == 1:
+            return [ExtractTreeSortFilterProxyModel.parse_date(date[0])]
+        elif len(date) == 2:
+            return [
+                ExtractTreeSortFilterProxyModel.parse_date(date[0]),
+                ExtractTreeSortFilterProxyModel.parse_date(date[1]),
+            ]
+        else:
+            raise argparse.ArgumentTypeError("Invalid date format. Can only accept two values.")
+
+    def get_parser(self):
+        """Add Extract view specific arguments to the parser."""
+        parser = super().get_parser()
+        parser.add_argument("--healthy", action="store_true", help="Match only healthy items.")
+        parser.add_argument("--unhealthy", action="store_true", help="Match only unhealthy items.")
+        parser.add_argument(
+            "--last-modified",
+            type=ExtractTreeSortFilterProxyModel.valid_date_range,
```

Comment:
```suggestion
            type=self.valid_date_range,
```

This is more pythonic I think + allows overiding